tracer/src/Datadog.Trace/Agent/MessagePack/SpanMessagePackFormatter.cs

```diff
@@ -164,17 +164,15 @@ private int WriteTags(ref byte[] bytes, int offset, Span span, ITags tags, ITagP
             {
                 // write trace-level string tags
                 var traceTags = span.Context.TraceContext?.Tags;
+
                 if (traceTags != null)
                 {
-                    lock (traceTags)
```

Comment:
Was the removal of this lock intentional?

Reply:
Uhm, kind of. I moved all the locking inside `TraceTagCollection` using a private field, but I forgot I had to figure something out for this case. Thanks.

Reply:
I refactored iterating into `ToArray()` which returns a copy that won't be modified.